dlog/commitment/src/srs.rs

```diff
@@ -70,9 +77,27 @@ where
         self.g.len()
     }
 
-    // This function creates SRS instance for circuits up to depth d
-    //      depth: maximal depth of SRS string
-    //      size: circuit size
+    /// Compute commitments to the lagrange basis corresponding to the given domain and
+    /// cache them in the SRS
+    pub fn add_lagrange_basis(&mut self, domain: D<G::ScalarField>) {
+        let n = domain.size();
+        if n > self.g.len() {
+            panic!("add_lagrange_basis: Domain size {} larger than SRS size {}", n, self.g.len());
+        }
+
+        if self.lagrange_bases.contains_key(&n) {
+            return;
+        }
+
+        let mut lg: Vec<<G as AffineCurve>::Projective> =
+            self.g[0..n].iter().map(|g| g.into_projective()).collect();
+        domain.ifft_in_place(&mut lg);
+
+        <G as AffineCurve>::Projective::batch_normalization(lg.as_mut_slice());
+        self.lagrange_bases.insert(n, lg.iter().map(|g| g.into_affine()).collect());
```

Comment:
I'm not sure I really understand what is happening here tbh. Also, a test would be nice to make sure that committing in evaluation forms is equivalent to committing in coeff form